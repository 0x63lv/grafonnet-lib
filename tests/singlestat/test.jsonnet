local grafana = import "grafonnet/grafana.libsonnet";
local singlestat = grafana.singlestat;

{
    basic: singlestat.new("single"),
    target:
        singlestat.new("targets")
        + singlestat.addTarget("target1")
        + singlestat.addTarget("target2")
        + singlestat.addTarget("target3")
    ,
    advanced: singlestat.new(
        "adv",
        format="s",
        interval="10s",
        repeatDirection="v",
        height="42px",
        prefixFontSize="10%",
        postfixFontSize="15%",
        prefix="before",
        postfix="after",
        repeat="interval",
        description="mystat",
        min_span=6,
        decimals=2,
        datasource="prom1",
        span=1,
        transparent=true,
        valueFontSize="50%",
        valueName="current",
        mappingType="range",
        valueMaps=["foo"],
        colors=[
            "rgba(237, 129, 40, 0.89)",
            "#d44a3a",
            "#299c46",
        ],
        colorBackground=true,
        colorValue=true,
        thresholds="1,2",
        rangeMaps=["bar"],
    ),
    value: singlestat.new(
        "vv",
        mappingType="value",
    ),
}
