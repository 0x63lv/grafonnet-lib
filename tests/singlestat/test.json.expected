{
   "advanced": {
      "cacheTimeout": null,
      "colorBackground": false,
      "colorValue": false,
      "colors": [
         "#299c46",
         "rgba(237, 129, 40, 0.89)",
         "#d44a3a"
      ],
      "datasource": "prom1",
      "format": "s",
      "gauge": {
         "maxValue": 100,
         "minValue": 0,
         "show": false,
         "thresholdLabels": false,
         "thresholdMarkers": true
      },
      "id": 1,
      "interval": "10s",
      "links": [ ],
      "mappingType": 1,
      "mappingTypes": [
         {
            "name": "value to text",
            "value": 1
         },
         {
            "name": "range to text",
            "value": 2
         }
      ],
      "maxDataPoints": 100,
      "nullPointMode": "connected",
      "nullText": null,
      "postfix": "",
      "postfixFontSize": "50%",
      "prefix": "",
      "prefixFontSize": "50%",
      "rangeMaps": [
         {
            "from": "null",
            "text": "N/A",
            "to": "null"
         }
      ],
      "span": 1,
      "sparkline": {
         "fillColor": "rgba(31, 118, 189, 0.18)",
         "full": false,
         "lineColor": "rgb(31, 120, 193)",
         "show": false
      },
      "tableColumn": "",
      "targets": [ ],
      "thresholds": "",
      "title": "adv",
      "type": "singlestat",
      "valueFontSize": "50%",
      "valueMaps": [
         {
            "op": "=",
            "text": "N/A",
            "value": "null"
         }
      ],
      "valueName": "current"
   },
   "basic": {
      "cacheTimeout": null,
      "colorBackground": false,
      "colorValue": false,
      "colors": [
         "#299c46",
         "rgba(237, 129, 40, 0.89)",
         "#d44a3a"
      ],
      "datasource": null,
      "format": "none",
      "gauge": {
         "maxValue": 100,
         "minValue": 0,
         "show": false,
         "thresholdLabels": false,
         "thresholdMarkers": true
      },
      "interval": null,
      "links": [ ],
      "mappingType": 1,
      "mappingTypes": [
         {
            "name": "value to text",
            "value": 1
         },
         {
            "name": "range to text",
            "value": 2
         }
      ],
      "maxDataPoints": 100,
      "nullPointMode": "connected",
      "nullText": null,
      "postfix": "",
      "postfixFontSize": "50%",
      "prefix": "",
      "prefixFontSize": "50%",
      "rangeMaps": [
         {
            "from": "null",
            "text": "N/A",
            "to": "null"
         }
      ],
      "span": 12,
      "sparkline": {
         "fillColor": "rgba(31, 118, 189, 0.18)",
         "full": false,
         "lineColor": "rgb(31, 120, 193)",
         "show": false
      },
      "tableColumn": "",
      "targets": [ ],
      "thresholds": "",
      "title": "single",
      "type": "singlestat",
      "valueFontSize": "80%",
      "valueMaps": [
         {
            "op": "=",
            "text": "N/A",
            "value": "null"
         }
      ],
      "valueName": "avg"
   },
   "target": {
      "cacheTimeout": null,
      "colorBackground": false,
      "colorValue": false,
      "colors": [
         "#299c46",
         "rgba(237, 129, 40, 0.89)",
         "#d44a3a"
      ],
      "datasource": null,
      "format": "none",
      "gauge": {
         "maxValue": 100,
         "minValue": 0,
         "show": false,
         "thresholdLabels": false,
         "thresholdMarkers": true
      },
      "interval": null,
      "links": [ ],
      "mappingType": 1,
      "mappingTypes": [
         {
            "name": "value to text",
            "value": 1
         },
         {
            "name": "range to text",
            "value": 2
         }
      ],
      "maxDataPoints": 100,
      "nullPointMode": "connected",
      "nullText": null,
      "postfix": "",
      "postfixFontSize": "50%",
      "prefix": "",
      "prefixFontSize": "50%",
      "rangeMaps": [
         {
            "from": "null",
            "text": "N/A",
            "to": "null"
         }
      ],
      "span": 12,
      "sparkline": {
         "fillColor": "rgba(31, 118, 189, 0.18)",
         "full": false,
         "lineColor": "rgb(31, 120, 193)",
         "show": false
      },
      "tableColumn": "",
      "targets": [
         "target1",
         "target2",
         "target3"
      ],
      "thresholds": "",
      "title": "targets",
      "type": "singlestat",
      "valueFontSize": "80%",
      "valueMaps": [
         {
            "op": "=",
            "text": "N/A",
            "value": "null"
         }
      ],
      "valueName": "avg"
   }
}
